services/agents/registry.go

```diff
@@ -408,6 +414,33 @@ func (r *Registry) SendSetStateRequest(ctx context.Context, pmmAgentID string) {
 	l.Infof("SetState response: %+v.", res)
 }
 
+func (r *Registry) sendAgentMetadata(stream agentpb.Agent_ConnectServer, agentID string) error {
+	ctx := stream.Context()
+	l := logger.Get(ctx)
+
+	// TODO: Refactor. Use repository or service to get an agent.
+	row := &models.Agent{AgentID: agentID}
+	if err := r.db.Reload(row); err != nil {
+		if err == reform.ErrNoRows {
+			return status.Errorf(codes.NotFound, "No Agent with ID %q.", agentID)
+		}
+		return errors.Wrap(err, "failed to find agent")
+	}
+	if row.AgentType != models.PMMAgentType {
+		return status.Errorf(codes.NotFound, "No pmm-agent with ID %q.", agentID)
+	}
```

Comment:
I think it is better to return Node ID from `authenticate` call – it already works with models, it will be easier to refactor later.